### PR TITLE
Combine zebra and panda FGS variants and tests

### LINKED ISSUE #1432: Make a top level flyscan_xray_centre_plan
We're now in a position where we can create a top level `flyscan_xray_centre_plan` which calls parts from `panda_flyscan_xray_centre_plan` and `zebra_flyscan_xray_centre_plan`, depending on the `use_panda` parameter. This will be a lot cleaner and remove quite a bit of duplicated code.

Differences between panda and zebra scans:

- They use different tidy-up-plans
- Panda scan does a check on the smargon velocity (although there was some debate as to whether this check is in the right place)
- Different FGS params
- Panda scan does an extra device set up (for the panda)
- Panda scan creates its directory just before setup

The best way to do this:
`panda_flyscan_xray_centre_plan` and `zebra_flyscan_xray_centre_plan` should contain small functions for the bits that differ, eg they both have a `setup_devices_for_gridscan`, and `tidy_up_devices` which do different things. Then, the top level plan can have a single check at the beginning to see which type of gridscan it's doing, and then import the functions from the correct place.

## Acceptance Criteria

- `panda_flyscan_xray_centre_plan` and `zebra_flyscan_xray_centre_plan` reduced to a few functions which differentiate them
- Top level plan created, and only needs to do one check to see whether it's doing the panda or zebra version

### PR DESCRIPTION
**Pull request by dperl-dls**
Fixes #1432 

Merges the normal and panda fgs variants and their tests

### To test:
1. Run tests

### CODE REVIEW

File: src/hyperion/experiment_plans/flyscan_xray_centre_plan.py [hunk 1]
```diff
@@ -246,10 +277,17 @@ def run_gridscan(
...
-    yield from set_flyscan_params(fgs_motors, parameters.FGS_params)
+    if features.use_panda_for_gridscan:
+        yield from set_flyscan_params(
+            fgs_motors := fgs_composite.panda_fast_grid_scan,
+            parameters.panda_FGS_params,
+        )
+    else:
+        yield from set_flyscan_params(
+            fgs_motors := fgs_composite.zebra_fast_grid_scan, parameters.FGS_params
+        )
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/hyperion/experiment_plans/flyscan_xray_centre_plan.py]: I really disagree, I think doing that is over-engineered. eventually if we have that many variations either they will have enough in common that there won't be too many checks, or they will have little enough in common that they need to be different plans anyway. nonetheless, I've done as you asked.

[the remark this answers is not in the document]
Comment (olliesilvester) [reply src/hyperion/experiment_plans/flyscan_xray_centre_plan.py]: Hmm, fair enough, maybe I'm thinking ahead too much. Hard to know what future variations of flyscan will look like.

**Review by olliesilvester — COMMENTED** [review]
I didn't get a chance yet to look at the tests or the config service, the rest of it looks nice though! All the config stuff will need to be merged first, so I won't approve yet, but there's a few comments

**Review by olliesilvester — COMMENTED** [review]
Tests mostly look good, but I've spotted
`test_if_smargon_speed_over_limit_then_log_error`
`test_tidy_up_plans_disable_panda_and_zebra`
are missing. The latter of these is not a particularly useful test, but the smargon one has saved me before

File: src/hyperion/experiment_plans/flyscan_xray_centre_plan.py [hunk 2]
```diff
@@ -215,171 +363,191 @@ def do_fgs():
...
+    yield from bps.unstage(fgs_composite.zocalo, group=group, wait=wait)
 
-    LOGGER.info("Starting grid scan")
-    yield from bps.stage(
-        fgs_composite.zocalo, group=ZOCALO_STAGE_GROUP
-    )  # connect to zocalo and make sure the queue is clear
-    yield from run_gridscan(fgs_composite, parameters)
 
-    LOGGER.info("Grid scan finished, getting results.")
+def _zebra_tidy(fgs_composite: FlyScanXRayCentreComposite):
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/hyperion/experiment_plans/flyscan_xray_centre_plan.py:478]: I don't really think they're generally applicable enough for that

File: src/hyperion/experiment_plans/flyscan_xray_centre_plan.py [hunk 3]
```diff
@@ -107,59 +129,185 @@ def create_devices(context: BlueskyContext) -> FlyScanXRayCentreComposite:
...
+    # We get the initial motor positions so we can return to them on zocalo failure
+    initial_xyz = np.array(
+        [
+            (yield from bps.rd(fgs_composite.sample_motors.x)),
+            (yield from bps.rd(fgs_composite.sample_motors.y)),
+            (yield from bps.rd(fgs_composite.sample_motors.z)),
+        ]
+    )
+
+    yield from feature_controlled.extra_setup(fgs_composite, parameters, initial_xyz)
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply src/hyperion/experiment_plans/flyscan_xray_centre_plan.py:198]: yeah good point

**Review by olliesilvester — APPROVED** [review]
Nice! One comment but looks good

**Pull request merged**